packages/optimizer/lib/transformers/OptimizeAssets.js

```diff
@@ -151,28 +152,84 @@ class OptimizeImages {
     this.maxSrcsetValues = MAX_SRCSET_VALUE_COUNT;
   }
 
-  async transform(root) {
-    if (!this.imageOptimizer) {
-      return;
-    }
+  async transform(root, runtimeParams) {
     const html = firstChildByTag(root, 'html');
     const body = firstChildByTag(html, 'body');
+    const {baseDomain, rewriteUrlsToAmpCache} = runtimeParams;
 
     let node = body;
     const imageOptimizationPromises = [];
-    while (node !== null) {
+    while (node) {
       if (node.tagName === 'template') {
         node = skipNodeAndChildren(node);
       } else {
-        if (node.tagName === 'amp-img') {
+        if (node.tagName === 'amp-img' && this.imageOptimizer) {
           imageOptimizationPromises.push(this.optimizeImage(node));
         }
+        if (rewriteUrlsToAmpCache) {
+          imageOptimizationPromises.push(
+            this.rewriteHosts(node, rewriteUrlsToAmpCache, baseDomain)
+          );
+        }
+
         node = nextNode(node);
       }
     }
     return Promise.all(imageOptimizationPromises);
   }
 
+  async _replaceUrl(url, host) {
+    return createCacheUrl(host, url);
+  }
+
+  async rewriteHost(src, host, baseDomain) {
+    if (!src) {
+      return;
+    }
+    if (src.startsWith('/')) {
```

Comment:
do you want to support relative urls for parent/sibling directories? e.g. `../../foo.jpg`